### ISSUE TITLE
fail in https: was loaded over HTTPS but requested an insecure stylesheet/script

### ISSUE THREAD

**Issue by fnoceda** [DarkaOnLine/SwaggerLume#97]
Hi, I install and configure and use this package in dev mode successful but my produccion server is in https, and i installed an generate the documentation, but in the url in console show this error 

[The page https://example.com/api/documentation was loaded over HTTPS but requested an insecure stylesheet/script was loaded over HTTPS, but requested an insecure ]

And the page don't load anything.

Please Help me, 
Best.

**Comment by maitrepylos** [comment]
Hello, we have the same problem, have some ideas on how to solve it.

**Comment by fnoceda** [comment]
Hi, mi solution is very ugly, but work, jaja, i edited the index.html and replace the url auto generated by the same url but use https in the url instead http.

**Comment by fnoceda** [comment]
this is the final result, this file is in.   public/swagger-ui-assets/index.html 



<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Swagger Lume API - TEST</title>
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,700|Source+Code+Pro:300,600|Titillium+Web:400,600,700" rel="stylesheet">
    <link rel="stylesheet" type="text/css" href="https://yourdomain.com/esalud-api/swagger-ui-assets/swagger-ui.css?v=66660759c894bca78c4cfd4cff53861c" >
    <link rel="icon" type="image/png" href="https://yourdomain.com/esalud-api/swagger-ui-assets/favicon-32x32.png?v=f5a27bead4774247ac6bdf7b7d461720" sizes="32x32" />
    <link rel="icon" type="image/png" href="https://yourdomain.com/esalud-api/swagger-ui-assets/favicon-16x16.png?v=0d05b2bf00619c78d6df7c1eeacac339" sizes="16x16" />
    <style>
        html
        {
            box-sizing: border-box;
            overflow: -moz-scrollbars-vertical;
            overflow-y: scroll;
        }
        *,
        *:before,
        *:after
        {
            box-sizing: inherit;
        }

        body {
            margin:0;
            background: #fafafa;
        }
    </style>
</head>

<body>

<svg xmlns="https://www.w3.org/2000/svg" xmlns:xlink="https://www.w3.org/1999/xlink" style="position:absolute;width:0;height:0">
    <defs>
        <symbol viewBox="0 0 20 20" id="unlocked">
            <path d="M15.8 8H14V5.6C14 2.703 12.665 1 10 1 7.334 1 6 2.703 6 5.6V6h2v-.801C8 3.754 8.797 3 10 3c1.203 0 2 .754 2 2.199V8H4c-.553 0-1 .646-1 1.199V17c0 .549.428 1.139.951 1.307l1.197.387C5.672 18.861 6.55 19 7.1 19h5.8c.549 0 1.428-.139 1.951-.307l1.196-.387c.524-.167.953-.757.953-1.306V9.199C17 8.646 16.352 8 15.8 8z"></path>
        </symbol>

        <symbol viewBox="0 0 20 20" id="locked">
            <path d="M15.8 8H14V5.6C14 2.703 12.665 1 10 1 7.334 1 6 2.703 6 5.6V8H4c-.553 0-1 .646-1 1.199V17c0 .549.428 1.139.951 1.307l1.197.387C5.672 18.861 6.55 19 7.1 19h5.8c.549 0 1.428-.139 1.951-.307l1.196-.387c.524-.167.953-.757.953-1.306V9.199C17 8.646 16.352 8 15.8 8zM12 8H8V5.199C8 3.754 8.797 3 10 3c1.203 0 2 .754 2 2.199V8z"/>
        </symbol>

        <symbol viewBox="0 0 20 20" id="close">
            <path d="M14.348 14.849c-.469.469-1.229.469-1.697 0L10 11.819l-2.651 3.029c-.469.469-1.229.469-1.697 0-.469-.469-.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-.469-.469-.469-1.228 0-1.697.469-.469 1.228-.469 1.697 0L10 8.183l2.651-3.031c.469-.469 1.228-.469 1.697 0 .469.469.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c.469.469.469 1.229 0 1.698z"/>
        </symbol>

        <symbol viewBox="0 0 20 20" id="large-arrow">
            <path d="M13.25 10L6.109 2.58c-.268-.27-.268-.707 0-.979.268-.27.701-.27.969 0l7.83 7.908c.268.271.268.709 0 .979l-7.83 7.908c-.268.271-.701.27-.969 0-.268-.269-.268-.707 0-.979L13.25 10z"/>
        </symbol>

        <symbol viewBox="0 0 20 20" id="large-arrow-down">
            <path d="M17.418 6.109c.272-.268.709-.268.979 0s.271.701 0 .969l-7.908 7.83c-.27.268-.707.268-.979 0l-7.908-7.83c-.27-.268-.27-.701 0-.969.271-.268.709-.268.979 0L10 13.25l7.418-7.141z"/>
        </symbol>


        <symbol viewBox="0 0 24 24" id="jump-to">
            <path d="M19 7v4H5.83l3.58-3.59L8 6l-6 6 6 6 1.41-1.41L5.83 13H21V7z"/>
        </symbol>

        <symbol viewBox="0 0 24 24" id="expand">
            <path d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"/>
        </symbol>

    </defs>
</svg>

<div id="swagger-ui"></div>

<script src="https://yourdomain.com/esalud-api/swagger-ui-assets/swagger-ui-bundle.js?v=e4f6298891c77e5d4a053bf9214ece86"> </script>
<script src="https://yourdomain.com/esalud-api/swagger-ui-assets/swagger-ui-standalone-preset.js?v=957e891dcbba9aae89b0d3c6690cf1f9"> </script>
<script>
    window.onload = function() {
        // Build a system
        const ui = SwaggerUIBundle({
            dom_id: '#swagger-ui',

            url: "https://yourdomain.com/esalud-api/docs",
            operationsSorter: null,
            configUrl: null,
            validatorUrl: null,
            oauth2RedirectUrl: "https://yourdomain.com/esalud-api/api/oauth2-callback",

            presets: [
                SwaggerUIBundle.presets.apis,
                SwaggerUIStandalonePreset
            ],

            plugins: [
                SwaggerUIBundle.plugins.DownloadUrl
            ],

            layout: "StandaloneLayout"
        })

        window.ui = ui
    }
</script>
</body>

</html>

**Comment by maitrepylos** [comment]
Ok thx.
Here is the solution I have put in place

app/Http/Helpers/Helper.php :

`<?php
use SwaggerLume\Exceptions\SwaggerLumeException;

if (!function_exists('swagger_lume_asset')) {
    /**
     * Returns asset from swagger-ui composer package.
     *
     * @param $asset string
     *
     * @return string
     * @throws \SwaggerLume\Exceptions\SwaggerLumeException
     */
    function swagger_lume_asset($asset): string
    {
        $file = swagger_ui_dist_path($asset);

        if (!file_exists($file)) {
            throw new SwaggerLumeException(sprintf('Requested L5 Swagger asset file (%s) does not exists', $asset));
        }

        return route('swagger-lume.asset', ['asset' => $asset, 'v' => md5($file)], app('request'));
    }
}`

boostrap/app.php : 
`<?php

require __DIR__ . '/../app/Helpers/Helpers.php';
require_once __DIR__.'/../vendor/autoload.php';
....
`

**Comment by maitrepylos** [comment]
boostrap/app.php : 

```php
<?php

require __DIR__ . '/../app/Helpers/Helpers.php';
require_once __DIR__.'/../vendor/autoload.php';
....
```

**Comment by chrislawson** [comment]
Ran into the same problem because my app server is behind a load balancer which converts inbound https requests to http.

Added the option to override with env var. 

`return route('swagger-lume.asset', ['asset' => $asset, 'v' => md5($file)], env('APP_LOAD_BALANCED', false));`

**Comment by andrey-helldar** [comment]
```
dd(
    route('swagger-lume.asset', ['asset' => $asset, 'v' => md5($file)], app('request')->secure()),
    route('swagger-lume.asset', ['asset' => $asset, 'v' => md5($file)], app('request')),
    route('swagger-lume.asset', ['asset' => $asset, 'v' => md5($file)])
);
```

![image](https://user-images.githubusercontent.com/10347617/117445070-f0f19a80-af42-11eb-8681-d468c60a37c6.png)

**Comment by mtracz** [comment]
The easiest way without these dirty hacks is to use fideloper package for trusted proxy.
https://github.com/fideloper/TrustedProxy
https://laravel.com/docs/8.x/requests#configuring-trusted-proxies

**Comment by ComradePashka** [comment]
Hi!

I have same issue and I added TrustedProxy middleware as suggested.

But now even tho I can see proper `route('swagger-lume.docs');` inside artisan tinker direclty on server (with https scheme) however it still passing http URL into swagger blade template!!

I also found parameter `proxy` in the `config\swagger-lume.php` but I didn't find any usege of this value except `tests\GeneratorTest.php` file, seems it is not used anyhow anywhere at all.


P.S. I have deployment for dev into bare k8s cluster and for test/prod into AWS so all of them are using load balancer and SSL certificates for every domain name is managed by cloudflare.
